lib/_http_server.js

```diff
@@ -281,7 +273,7 @@ function connectionListener(socket) {
   if (self.timeout)
     socket.setTimeout(self.timeout);
   socket.on('timeout', function() {
-    var req = socket.parser && socket.parser.incoming;
+    var req = parser && parser.incoming;
```

Comment:
It seems like this would be a good time to use function hoisting so that all uses appear after definition.